pkg/console/operator/sync_v400.go

```diff
@@ -200,13 +200,27 @@ func SyncConfigMap(co *ConsoleOperator, consoleConfig *v1alpha1.Console, rt *rou
 // there is nothing special about our service, so no additional error handling is needed here.
 func SyncService(co *ConsoleOperator, consoleConfig *v1alpha1.Console) (*corev1.Service, bool, error) {
 	logrus.Printf("validating console service...")
-	svc, svcChanged, svcErr := resourceapply.ApplyService(co.serviceClient, servicesub.DefaultService(consoleConfig))
-	if svcErr != nil {
-		logrus.Errorf("%q: %v \n", "service", svcErr)
-		return nil, false, svcErr
+	svc, getErr := co.serviceClient.Services(controller.TargetNamespace).Get(servicesub.Stub().Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(getErr) {
+		_, svcChanged, svcErr := resourceapply.ApplyService(co.serviceClient, servicesub.DefaultService(consoleConfig))
+		return nil, svcChanged, fmt.Errorf("service not found, creating new service, create err = %v", svcErr)
+	}
+	if getErr != nil {
+		logrus.Errorf("%q: %v \n", "service", getErr)
+		return nil, false, getErr
+	}
+
+	if validatedService, changed := servicesub.Validate(svc); changed {
+		if _, err := co.serviceClient.Services(controller.TargetNamespace).Update(validatedService); err != nil {
```

Comment:
We cannot use `resourceapply.ApplyService()` here as it does not thoroughly check the `Service.Spec`.  If used, it will simply return the `existing` service & never actually issue an `Update` (our loop will `sync` forever). 

`resourceapply.ApplyService()` does work as a model for `SyncService()`, however.

Reply:
Can you expand on what specifically it doesn't check that we need?

Reply:
It checks if `selector` matches, `type` matches, and does `EnsureObjectMeta` (which sets `namespace`, `name` & does a `merge` of `labels` and `annotations`).  

So it does not look at `ports`, `type`, or `sessionAffinity`. 

This could potentially be an upstream PR to resolve if we want to do that.  I can add a comment here.

Reply:
Yup.  Shall we keep this for now while I work on the upstream?  I can swap back to `ApplyService` when it merges & I update dependencies.

Reply:
Made a comment to doc this. I'd prefer to get the operator functionality correct now, and then move the logic around, I think.

Reply:
Thinking we may want this validation now?